wolfcrypt/src/ecc.c

```diff
@@ -8042,6 +8042,8 @@ static int wc_ecc_check_r_s_range(ecc_key* key, mp_int* r, mp_int* s)
    res         Result of signature, 1==valid, 0==invalid
    key         The corresponding public ECC key
    return      MP_OKAY if successful (even if the signature is not valid)
+               and other negative values on error. Caller should check
```

Comment:
The `wc_ecc_verify_hash` has the `res` parameter for this. If someone provides valid r/s values for the curve, but does not lie on the key's curve then we return `MP_OKAY` and make sure `res = 0`. This is not considered a failure that returns a negative error code.